asv_bench/benchmarks/groupby.py

```diff
@@ -556,130 +529,88 @@ def time_groupby_extra_cat_nosort(self):
         self.df_extra_cat.groupby('a', sort=False)['b'].count()
 
 
-class groupby_period(object):
+class Datelike(object):
     # GH 14338
     goal_time = 0.2
+    params = [period_range, date_range, partial(date_range, tz='US/Central')]
```

Comment:
Can you use strings here and then make a dict string -> function in setup? 
(now you don't get nice param value names in the output)